### ISSUE TITLE
Issues in tf.keras.utils.timeseries_dataset_from_array

### ISSUE THREAD

**Issue by gnoejh** [keras-team/keras#17410]
Issue 1:
The line 111 in timeseries_dataset_from_array [https://github.com/keras-team/keras/blob/v2.11.0/keras/utils/timeseries_dataset.py#L24-L266] must be corrected
from     input_data = data[:-10]
to input_data = data[:-1].
The input data can be used all except the last one to build dataset for predicted one time step forward.
 In general, for predicting delay time steps ahead with sequence_length, the code must be used with
    tf.keras.utils.timeseries_dataset_from_array(
        data = data[:-delay],
        targets = target[sequence_length+delay-1:]
        sequence_length,
        sequence_stride=1,
        sampling_rate=1,
        batch_size=128,
        shuffle=False,
        seed=None,
        start_index=None,
        end_index=None
    )
Issue 2: shuffle must be modified so that the shuffle occurs in batch level, not changing the chlonogical order of the sequence. To avoid this problem instead of shuffle = True, we must use the method, train_dataset.shuffle(10) for example.

**Comment by sushreebarsa** [comment]
@gnoejh Thank you for reporting this issue. We can create the PR to fix this. Could you elaborate on the second issue and the solution you want to propose? I have raised a PR to fix the first issue, please have a look and confirm if it is fine?
Thank you!

**Comment by sushreebarsa** [comment]
@gnoejh Could you please refer this [comment](https://github.com/keras-team/keras/pull/17411#issuecomment-1399408794), I also didn't face the error while executing the existing example. Could you please elaborate on the issue reported and required changes?
Thank you!
